src/modules/data/sound.c

```diff
@@ -68,8 +68,10 @@ static uint32_t lovrSoundReadMp3(Sound* sound, uint32_t offset, uint32_t count,
   }
 
   uint32_t channels = lovrSoundGetChannelCount(sound);
-  size_t samples = mp3dec_ex_read(sound->decoder, data, count * channels);
-  uint32_t frames = samples / channels;
+  uint32_t readSamples = (uint32_t)(count * channels);
+  lovrCheck(readSamples/count == channels, "Cannot request more than 2^32-1 samples of sound data at a time"); // Check for overflow
```

Comment:
It's not safe to throw in this function since it can run in the audio thread.  It could cap the number of frames it tries to read and return a smaller value, which would cause the mixer to retry with the remaining number of frames.